test/client_auth/main.d

```diff
@@ -0,0 +1,86 @@
+/*******************************************************************************
+
+    Test for loading client auth name/key from a file.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH. All rights reserved
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+
+*******************************************************************************/
+
+module test.client_auth.main;
+
+import test.neo.client.Client;
+import swarm.neo.authentication.CredentialsFile;
+import ocean.transition;
+import ocean.core.Test;
+import ocean.sys.ErrnoException;
+import ocean.io.device.File;
+import Path = ocean.io.Path;
+import ocean.io.select.EpollSelectDispatcher;
+import ocean.text.util.StringC;
+import core.sys.posix.unistd;
+import ocean.stdc.posix.stdlib : mkdtemp;
+
+void main ( )
+{
+    // Exception to throw on failure of C functions.
+    auto e = new ErrnoException;
+
+    // Get current working dir.
+    char[4096] old_cwd;
+    e.enforceRetPtr!(getcwd).call(old_cwd.ptr, old_cwd.length);
+
+    // Create a temp dir.
```

Comment:
You could use `DirectorySandbox` utility from the ocean: https://github.com/sociomantic-tsunami/ocean/blob/v2.x.x/src/ocean/util/test/DirectorySandbox.d

```
auto sandbox = DirectorySandbox.create();
scope (exit) sandbox.remove();
```